plasticparcels/constructors.py

```diff
@@ -202,63 +202,32 @@ def create_fieldset(settings):
+def create_particleset(fieldset, settings, release_locations):
+    """ A constructor method to create a Parcels.ParticleSet for a plasticparcels simulation
 
     Parameters
     ----------
     fieldset :
         A Parcels.FieldSet object
     settings :
         A dictionary of model settings, simulation settings, and release settings and plastic-type settings
+    release_locations :
+        A dictionary of release locations for particles
 
     Returns
     -------
     particleset
         A parcels.ParticleSet object
     """
 
-    # Load release type information
-    release_type = settings['release']['initialisation_type']  # TODO: MAKE THIS PART BETTER!
-
-    release_quantity_names = {
-        'coastal': 'MPW_Cell',
-        'rivers': 'Emissions',
-        'fisheries': 'fishing_hours',
-        'global_concentrations': 'Concentration'
-    }
-    release_quantity_name = release_quantity_names[release_type]
-
-    particle_locations = pd.read_csv(settings['release_maps'][release_type])
-
-    # Select specific continent/region/subregion/country/economic status if applicable:
-    if 'continent' in settings['release'].keys():
-        particle_locations = particle_locations[particle_locations['Continent'] == settings['release']['continent']]
-    if 'region' in settings['release'].keys():
-        particle_locations = particle_locations[particle_locations['Region'] == settings['release']['region']]
-    if 'subregion' in settings['release'].keys():
-        particle_locations = particle_locations[particle_locations['Subregion'] == settings['release']['subregion']]
-    if 'country' in settings['release'].keys():
-        particle_locations = particle_locations[particle_locations['Country'] == settings['release']['country']]
-    if 'economicstatus' in settings['release'].keys():
-        particle_locations = particle_locations[particle_locations['Economic status'] == settings['release']['economicstatus']]
-    if 'concentration_type' in settings['release'].keys():
-        particle_locations = particle_locations[particle_locations['ConcentrationType'] == settings['release']['concentration_type']]
-
-    particle_locations = particle_locations.groupby(['Longitude', 'Latitude'])[release_quantity_name].agg('sum').reset_index()
-    particle_locations = particle_locations[particle_locations[release_quantity_name] > 0]
-
-    release_locations = {'lons': particle_locations['Longitude'],
-                         'lats': particle_locations['Latitude'],
-                         'plastic_amount': particle_locations[release_quantity_name]}
-
     # Set the longitude, latitude, and plastic amount per particle
     lons = release_locations['lons']
     lats = release_locations['lats']
-    plastic_amounts = release_locations['plastic_amount']
+    if 'plastic_amount' in release_locations.keys():
+        plastic_amounts = release_locations['plastic_amount']
+    else:
+        plastic_amounts = np.zeros_like(lons)
```

Comment:
Would setting these plastic_amounts to `np.nan` not make more sense?

Reply:
I agree, that makes more sense (a zero might be assumed a true zero or a missing value, a NaN can only be interpreted as a missing value)

Reply:
I agree, that makes more sense (a zero might be assumed a true zero or a missing value, a NaN can only be interpreted as a missing value). Have updated now.